storybook/cypress/support/image-snapshot-voorbeeldpaginas.ts

```diff
@@ -0,0 +1,38 @@
+export function voorbeeldpaginaImageSnapshots(label: string, voorbeeldpaginas: string[], args?: string) {
+  describe(`Voorbeeldpagina image snapshots (${label})`, () => {
+    for (const id of voorbeeldpaginas) {
+      it(`matches image snapshot of ${id} (${label})`, () => {
+        cy.visit(`http://localhost:45000/iframe.html?id=${id}${args ? `&args=${args}` : ""}`);
+
+        cy.get("#root-inner .hydrated").should(pauze(($subject) => $subject.length > 0));
+
+        cy.matchImageSnapshot();
+      });
+    }
+  });
+}
+
+/**
+ * Pauze until the given assertion is true or the timeout is reached.
+ *
+ * The timeout in this function is a different (and lower) timeout than the Cypress should() timeout.
+ * pauze() works because Cypress will catch any errors thrown in the callback and retry the assertion command.
+ * If the assertion is still false after pauze()'s timeout, the test will continue because pauze() will stop
+ * throwing errors.
+ */
+function pauze<T extends JQuery<HTMLElement>>(assert: (subject: T) => boolean, { timeout = 1500 }: PauzeOptions = {}) {
```

Comment:
@iterox Nou kijk eens wat ik uit de krochten van Cypress heb weten te persen...

Reply:
Helaas zijn er nu verschillen. 🤷‍♂️

Reply:
😭

Reply:
Ik geef het op: https://github.com/dso-toolkit/dso-toolkit/issues/2694

Er is iets heel anders aan de hand dan async issues. Zelfs met een `.wait(5000)` krijg ik verschillen: Soms rendered het hele Header component niet, of met een verkeerd font. Dan staat boven een standaard font, en de rest van de pagina heeft Asap. Onbegrijpelijk.

Reply:
Weet jij nog waarom we de image-snapshot-header-fix css niet hebben meegenomen in de cypressStyling? Want idd zie ik veel headers die verschillend zijn. Draai nu even lokaal met deze css weer toegevoegd:

```scss
  dso-header dso-icon,
  dso-header svg.di,
  dso-header dso-dropdown-menu {
    visibility: hidden !important;
  }
```

Reply:
Draaien met bovenstaande css weer enabled en nog met de pauze-constructie maakt geen verschil. Veel verschillen dus. Draai ook de code in `image-snapshot-voorbeeldpaginas.ts` terug.

Reply:
En er zijn geen verschillen meer...

Reply:
Maar wel valen er nu 2 testen van de header omdat de image-header-fix van hierboven natuurlijk niet moet worden toegepast op de testen van de header zelf.

Reply:
Ik heb bovenstaande opgelost door in `image-snapshot-voorbeeldpaginas.ts` een query-parameter `imagesnapShotHeaderFix: true` mee te geven aan de `cy.visit`, welke dan in `commands.ts` in `Cypress.Commands.overwrite("visit", ...` aan `cypressStyling(options?.qs?.imagesnapShotHeaderFix);` wordt meegeven...

Reply:
Ik blijf tegen timing issues aanlopen. We tillen de voorbeeldpagina's uit de scope van dit issue en pakken dat op in #2694.